lib/driver.coffee

```diff
@@ -94,15 +64,19 @@ createExtDriverDisposer = (disk, offset, size) ->
 #     Promise.using createDriverFromFile(disk), (driver) ->
 # 	      console.log(driver)
 ###
-createDriverFromFile = (disk, offset, size) ->
-	detectFS(disk, offset, size)
-	.then (driver) ->
-		if driver == 'fatfs'
-			createFatDriverDisposer(disk, offset, size)
-		else if driver == 'node-ext2fs'
-			createExtDriverDisposer(disk, offset, size)
+createDriverFromFile = (disk, offset, size, type) ->
+	Promise.try ->
+		if type
+			if type == 0x83
+				createExtDriverDisposer(disk, offset, size)
+			else
+				createFatDriverDisposer(disk, offset, size)
 		else
-			throw new Error('Unsupported filesystem.')
+			createFatDriverDisposer(disk, offset, size)
```

Comment:
I added tests for mounting raw fat and ext partitions, it tests that createFatDriverDisposer throws an error if given an ext partition indirectly.